Upgrade version of mypy

zzz4zzz opened the pull request:
- Upgrade version of mypy
- Fix typing errors

lopuhin commented on eli5/keras/gradcam.py:
```diff
@@ -199,7 +199,7 @@ def _validate_target(target, output_shape):
         if not (0 <= target < output_nodes):
             raise ValueError('Prediction target index is ' 
-                             'outside the required range [0, {}). ',
+                             'outside the required range [0, {}). '
```
Thanks for the catch 👍

lopuhin commented on eli5/formatters/utils.py:
```diff
@@ -144,7 +144,7 @@ def tabulate(data,  # type: List[List[Any]]
 def format_weight(value):
     # type: (Real) -> str
-    return '{:+.3f}'.format(value)
+    return '{:+.3f}'.format(value)  # type: ignore
```
I see that this is due to https://github.com/python/mypy/issues/3186, maybe a better fix would be to use ``Union[int, float]`` here and below instead of the ``Real`` until that issue is fixed, especially since we use ``Real`` only in this module

zzz4zzz replied:
I changed `Real` to `float`. No need to use `Union[float, int]` ( https://www.python.org/dev/peps/pep-0484/#the-numeric-tower ).

lopuhin commented on eli5/formatters/text.py:
```diff
@@ -159,7 +159,7 @@ def _transition_features_lines(explanation):
         "",
         "Transition features:",
-        tabulate(tf.coef, headers=tf.class_names, showindex=tf.class_names,
+        tabulate(tf.coef, headers=tf.class_names, showindex=tf.class_names,  # type: ignore
```
Looks like mypy is confused by local ``tabulate`` import shadowing the global import, probably a better fix would be to move ``from tabulate import tabulate`` to the top level and import ``tabulate`` from utils as ``eli5_tabulate`` or something like this?

zzz4zzz replied:
Yes, good point. Changed.

zzz4zzz commented:
@lopuhin I've checked your comments and made fixes. Additionaly I've added [--ignore-missing-imports](https://mypy.readthedocs.io/en/latest/command_line.html#cmdoption-mypy-ignore-missing-imports) to mypy command line and cleaned up imports.

lopuhin commented:
Thanks @zzz4zzz , and ``--ignore-missing-imports`` is also a welcome addition. I'll check PR in more detail later today or tomorrow.

lopuhin commented:
Thanks for review @Gallaecio 👍
